### ISSUE TITLE
change plot limits

### ISSUE THREAD

**Issue by fmannucci** [dsavoiu/kafe2#114]
I need to change the limits of the plot to see the extrapolation of the best-fitting model and of its errors, but I cannot find how to do it, can you help? I understand I can use plot.customize() but I do not find a list of plot_types and keywords. Thanks.

**Comment by Citiroller** [comment]
There is currently no such feature afaik.

Technical ideas on how to implement this:
Every plot adapter (for XYFit, HistFit, etc) has a `x_range` property but no setter. There should be a setter (maybe in the plot adapter base) and this should be made accessible through the `plot.customize()` function.
The `x_range` is only used as supporting points when plotting the model and the error bar. By providing a setter the point density can be increased (this adds the possibility to fix the problem in #25 if needed).
Currently the `customize` function does only support additional kwargs though and not the x data used for plotting the error band. So there might need to be an additional plot property added for the `x_range` which is then passed through to the plot adapters.

**Comment by Citiroller** [comment]
As a current workaround you can always get the fit results and uncertainties and plot the model and error band directly in matplotlib.

**Comment by Citiroller** [comment]
Similiar to #85

**Comment by Citiroller** [comment]
@fmannucci can you please test this version https://github.com/dsavoiu/kafe2/tree/plot_range
You can now use:
```
plot.set_x_range((x_min, x_max))  # set range for all fits inside the plot
plot.set_x_range([(x_min1, x_max1), (x_min2, x_max2), ...])  # set the limits for each fit individually
```
Do you have any recommendations on a better function signature?

**Comment by Citiroller** [comment]
It's now been reworked to a property. I'm still deciding weather to use a property or setter function.
```
plot.x_range = (xmin, xmax)  # for single fit in plot, or all fits when multiple fits in plot
plot.x_range = [(xmin, xmax), (xmin2, xmax2), ...]  # for mutliple fits in plot
```
The same goes for y_range

**Comment by fmannucci** [comment]
Thanks very much, both for kafe1=2, a very useful tool, and for the extra work due to my question! I can test the software but I don't know how to install the new version, pip still refers to the previous one.

**Comment by JohannesGaessler** [comment]
`pip install kafe2` installs kafe2 release builds from pypi (there has not yet been a new release).
However, you can get the version currently on Gihub by installing kafe2 from a git repository.
To do this, use `pip install path/to/repository` after uninstalling the kafe2 version you got from pypi.

**Comment by fmannucci** [comment]
Thanks for the modifications, I tested them. 
plot.x_range works perfectly fine, it changes the range of both panels of plot.plot (I had to change "with_asymmetric_parameter_errors=True, with_ratio=True" into "asymmetric_parameter_errors=True, ratio=True". )
Also plot.y_range changes the ranges of both panels, but in this case this is not what is needed

**Comment by Citiroller** [comment]
Yes, the keywords have been renamed, to be the same as in `fit.report()` for asymmetric errors. Subsequently the with-Prefix was removed from the keywords. Please keep in mind that we're currently still in a beta state and things might change. Those changes will be included in the next release.
I see the problem, the ratio y-range is also influenced by the y_range setter. I'll take a look into it. I did not test that.

**Comment by Citiroller** [comment]
The y_range ration issue has been fixed in 78032fd44a540e4c28b0252deacad83c885f461e. I will add a feature to change the plot range on contour plots as well and then merge #116 into master once it's done. Those changes should be included in the next PyPi release.
